File: SYP.Web/Modules/Setting/NumberTemplates/NumberTemplatesForm.cs
using Serenity.ComponentModel;

namespace SYP.Setting.Forms;

[FormScript("Setting.NumberTemplates")]
[BasedOnRow(typeof(NumberTemplatesRow), CheckNames = true)]
public class NumberTemplatesForm
{
    [Category("Template Settings")]
    [DisplayName("Template Type"), EnumEditor]
    public global::_Ext.NumberTemplateType Type { get; set; }

    [Placeholder("Prefix (e.g., MUS-, URN-)")]
    public string Prefix { get; set; }

    [Placeholder("Leave blank to exclude date"), Hint(@"
Example Formats:
yyyy     => 2024
yy       => 24
yyyyMM   => 202401
yyyyMMdd => 20240115
MMyyyy   => 012024
")]
    public string DateFormat { get; set; }

    [Placeholder("Suffix (after date)")]
    public string Suffix { get; set; }

    [Placeholder("Serial number length (e.g., 5 => 00001)")]
    public int Length { get; set; }

    [DefaultValue(true)]
    public bool Active { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Setting/NumberTemplatesForm.ts
import { BooleanEditor, EnumEditor, initFormType, IntegerEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";
import { NumberTemplateType } from "./NumberTemplateType";

export interface NumberTemplatesForm {
    Type: EnumEditor;
    Prefix: StringEditor;
    DateFormat: StringEditor;
    Suffix: StringEditor;
    Length: IntegerEditor;
    Active: BooleanEditor;
}

export class NumberTemplatesForm extends PrefixedContext {
    static readonly formKey = 'Setting.NumberTemplates';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!NumberTemplatesForm.init) {
            NumberTemplatesForm.init = true;

            var w0 = EnumEditor;
            var w1 = StringEditor;
            var w2 = IntegerEditor;
            var w3 = BooleanEditor;

            initFormType(NumberTemplatesForm, [
                'Type', w0,
                'Prefix', w1,
                'DateFormat', w1,
                'Suffix', w1,
                'Length', w2,
                'Active', w3
            ]);
        }
    }
}

[NumberTemplateType]; // referenced types